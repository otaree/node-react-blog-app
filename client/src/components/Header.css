header {
    background-color: rgb(94, 172, 224);
    margin: 0;
    padding: 3px;
    width: 100vw;
    color: #888;
}

header nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

header nav div {
    margin-left: 10px;
    font-size: 20px;
    color: rgb(19, 19, 129);
    font-weight: bold;
}


header nav ul {
    display: flex;
    justify-content: flex-end;
    list-style: none;
}

header nav ul li {
    margin-right: 10px;
}

header nav ul li a {
    text-decoration: none;
    color: #555;
    font-size: 18px;
}

header nav ul li a:hover {
    color: #ddd;
}

header .active {
    color: white;
}