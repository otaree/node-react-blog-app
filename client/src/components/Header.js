import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';

const Header = (props) => {
    let ul = (
        <ul>
            <li><NavLink exact to="/" activeClassName="active">Home</NavLink></li>
            <li><NavLink to="/auth/signup" activeClassName="active">Auth</NavLink></li>
        </ul>
    );
    if (props.isAuth) {
        ul = (
            <ul>
                <li><NavLink exact to="/" activeClassName="active">Home</NavLink></li>
                <li><NavLink to="/blogs/create" activeClassName="active">Create</NavLink></li>
                <li><NavLink to="/logout" activeClassName="active">Logout</NavLink></li>
            </ul>
        );
    }
    return (
        <header>
            <nav>
                <div>Blog</div>
                {ul}
            </nav>
        </header>
    );
};

export default Header;
