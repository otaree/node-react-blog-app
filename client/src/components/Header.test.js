import React from 'react';
import { shallow } from 'enzyme';
import  Header  from './Header';

describe('Header', () => {
    const header = shallow(<Header />);

    it('renders properly', () => {
        expect(header).toMatchSnapshot();
    });
});