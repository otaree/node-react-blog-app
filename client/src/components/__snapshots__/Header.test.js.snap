// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav>
        <div>
                Blog
        </div>
        <ul>
                <li>
                        <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                exact={true}
                                to="/"
                        >
                                Home
                        </NavLink>
                </li>
                <li>
                        <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/auth/signup"
                        >
                                Auth
                        </NavLink>
                </li>
        </ul>
</nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Blog
</div>,
          <ul>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    exact={true}
                                    to="/"
                        >
                                    Home
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/auth/signup"
                        >
                                    Auth
                        </NavLink>
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Blog",
          },
          "ref": null,
          "rendered": "Blog",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                exact={true}
                                to="/"
                >
                                Home
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/auth/signup"
                >
                                Auth
                </NavLink>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  exact={true}
                  to="/"
>
                  Home
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Home",
                  "exact": true,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/auth/signup"
>
                  Auth
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Auth",
                  "to": "/auth/signup",
                },
                "ref": null,
                "rendered": "Auth",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav>
          <div>
                    Blog
          </div>
          <ul>
                    <li>
                              <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        exact={true}
                                        to="/"
                              >
                                        Home
                              </NavLink>
                    </li>
                    <li>
                              <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        to="/auth/signup"
                              >
                                        Auth
                              </NavLink>
                    </li>
          </ul>
</nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Blog
</div>,
            <ul>
              <li>
                            <NavLink
                                          activeClassName="active"
                                          ariaCurrent="true"
                                          exact={true}
                                          to="/"
                            >
                                          Home
                            </NavLink>
              </li>
              <li>
                            <NavLink
                                          activeClassName="active"
                                          ariaCurrent="true"
                                          to="/auth/signup"
                            >
                                          Auth
                            </NavLink>
              </li>
</ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Blog",
            },
            "ref": null,
            "rendered": "Blog",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    exact={true}
                                    to="/"
                  >
                                    Home
                  </NavLink>
</li>,
                <li>
                  <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/auth/signup"
                  >
                                    Auth
                  </NavLink>
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    exact={true}
                    to="/"
>
                    Home
</NavLink>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Home",
                    "exact": true,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/auth/signup"
>
                    Auth
</NavLink>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Auth",
                    "to": "/auth/signup",
                  },
                  "ref": null,
                  "rendered": "Auth",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
